source/common/upstream/outlier_detection_impl.cc

```diff
@@ -17,8 +17,8 @@ DetectorPtr DetectorImplFactory::createForCluster(Cluster& cluster,
   // Right now we don't support any configuration but in order to make the config backwards
   // compatible we just look for an empty object.
   if (cluster_config.hasObject("outlier_detection")) {
-    return DetectorImpl::create(cluster, dispatcher, runtime, ProdSystemTimeSource::instance_,
-                                event_logger);
+    return DetectorImpl::create(cluster, cluster_config.getObject("outlier_detection", true),
```

Comment:
don't "allow_empty" here (remove true param), the object must exist in this branch.